### PR TITLE
Added info about revoking compound privileges also revokes sub-privileges

### PR DESCRIPTION
**Pull request by Hunterness**
Should not be forwarded to 4.1 since it works differently there.

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 1]
```diff
@@ -224,20 +232,34 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
...
       query("REVOKE TRAVERSE ON GRAPH neo4j NODES Payments FROM regularUsers", ResultAssertions((r) => {
-        // TODO: revoke a privilege that exists
-        assertStats(r, systemUpdates = 0)
+        assertStats(r, systemUpdates = 2)
       })) {
         statsOnlyResultTable()
       }
+
+      p(
+        """Some privileges are compound privileges and contains sub-privileges, for example <<administration-security-administration-database-indexes, `INDEX MANAGEMENT`>> which covers `CREATE INDEX` and `DROP INDEX`.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala:249]: Should we mention that you can't revoke a sub-privilege when a compound version of the compound privilege exists?

Comment (craigtaverner) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala:249]: The tricky part is that in theory all compound privileges should have been split to leaves on creation. This is true for all that were created using Cypher, but not true for some of those created in the system graph initialization. So this problem is specific to system graph initialization privileges only, and perhaps needs to be described in the operations manual, not here where we focus on Cypher commands only (and there are no bugs).

**Review by craigtaverner — APPROVED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala:249]: `This is true for all that were created using Cypher` Except for ROLE MANAGEMENT :P

**Review by davidoliverSP2 — APPROVED** [review]
thanks @Hunterness 👍

**Pull request merged**